clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7520,9 +7520,24 @@ static void handleSYCLIntelDoublePumpAttr(Sema &S, Decl *D,
 
 /// Handle the [[intel::fpga_memory]] attribute.
 /// This is incompatible with the [[intel::fpga_register]] attribute.
+SYCLIntelMemoryAttr *
+Sema::MergeSYCLIntelMemoryAttr(Decl *D, const AttributeCommonInfo &AL,
```

Comment:
Do we really need this logic when attributes are applied on variables only? It might make sense to see how community handles this for other attributes. I understand the need for this when we have function attributes. I don't really see the need for it on variable attributes though

Reply:
Thank you @elizabethandrews and @premanandrao for reviews!

>>I understand the need for this when we have function attributes. I don't really see the need for it on variable attributes though

Yes, community uses this logic for function attributes only. I have checked this. So i have reverted my patch and aligned with community.